--- spectro/inst.c.new	2013-07-03 12:29:22.377610031 +0100
+++ spectro/inst.c	2013-07-03 12:30:04.883624886 +0100
@@ -1486,7 +1486,45 @@ static void hex2bin(char *buf, int len)
 
 
 
+#include "ucmm.h"
 
+ucmm_error
+ucmm_install_monitor_profile(
+	ucmm_scope scope,
+	unsigned char *edid,
+	int edid_len,
+	char *display_name,
+	char *profile)
+{
+	return ucmm_resource;
+}
+
+ucmm_error
+ucmm_uninstall_monitor_profile(
+	ucmm_scope scope,
+	unsigned char *edid,
+	int edid_len,
+	char *display_name,
+	char *profile)
+{
+	return ucmm_resource;
+}
+
+ucmm_error
+ucmm_get_monitor_profile(
+	unsigned char *edid,
+	int edid_len,
+	char *display_name,
+	char **profile)
+{
+	return ucmm_resource;
+}
+
+char *
+ucmm_error_string(ucmm_error erno)
+{
+	return "colord-not-found";
+}
 
 
 
